### PR TITLE
Change label path to be similar to repo and bundle.

### PR DESCRIPTION
**Pull request by kerneltime**
Change how labels are written to object stores.

Signed-off-by: Ritesh H Shukla <ritesh@oneconcern.com>

### CODE REVIEW

File: pkg/model/label.go [hunk 1]
```diff
@@ -39,12 +42,41 @@ func GetArchivePathPrefixToLabels(repo string, prefixes ...string) string {
...
+		}
+	}
+	for _, c := range label.Contributors {
+		if c.Name == "" {
+			return fmt.Errorf("name for contributor cannot be blank")
+		}
+		if c.Email == "" {
+			return fmt.Errorf("email for contributor cannot be blank")
+		}
+		// TODO: Is this sufficient validation?
```
Comment (fredbi) [inline pkg/model/label.go]: You may also use the more standard https://golang.org/pkg/net/mail/#AddressParser.Parse

File: pkg/core/label_list.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 const (
-	typicalLabelsNum = 1000 // default number of allocated slots for repos
+	typicalLabelsNum = 1000 // default number of allocated slots for labels
```
Comment (fredbi) [inline pkg/core/label_list.go]: :1st_place_medal:

File: pkg/model/label.go (hunk 1 above)
Comment (fredbi) [reply pkg/model/label.go]: Or skip this validation altogether, since email is assumed trusted from auth.

File: pkg/model/label.go [hunk 3]
```diff
@@ -39,12 +42,41 @@ func GetArchivePathPrefixToLabels(repo string, prefixes ...string) string {
 	return fmt.Sprint(getArchivePathToLabels(), repo+"/"+strings.Join(prefixes, "/"))
```
Comment (fredbi) [inline pkg/model/label.go]: this comment for more explanations: how are we supposed now to carry on with prefix-based search on labels?

Comment (kerneltime) [reply pkg/model/label.go]: I am not sure I understand this comment. The prefix should work as expected. Ref: https://github.com/oneconcern/datamon/blob/labels/cmd/datamon/cmd/cli_test.go#L609

Comment (fredbi) [reply pkg/model/label.go]: ok

**Comment by kerneltime** [comment]
@fredbi the bundle_list_test.go needs refactoring (not part of this test) to implement more stringent checks on path validation. (model/bundle.go)

**Pull request merged**